components/tikv_util/src/buffer_vec.rs

```diff
@@ -86,6 +86,20 @@ impl BufferVec {
         self.data.extend_from_slice(buffer.as_ref());
     }
 
+    /// Appends multiple buffer parts together as one buffer to the back.
+    #[inline]
+    pub fn push_together<T, I>(&mut self, buffer_parts: I)
```

Comment:
@nrc 

which better name do you suggest? I think `push_together` is strange.

Reply:
You can implement the `std::iter::Extend` trait instead.

Reply:
The `Extend` trait is already implemented for the non-together version, i.e. extending 5 `&[u8]`s results in 5 new elements. @BusyJay

Reply:
How about `concat_extend`?

Reply:
The naming LGTM.

Reply:
I agree `concat_extend` is good. Other alternatives: `collect_and_append` (from `Iterator::collect`), `flat_append` or `flat_push` (from `Iterator::flat_map`)